### PR TITLE
Tournament api http server

### PR DESCRIPTION
**Pull request by adamschachne**
Part 2 of the tournament api feature

Adds a FastAPI+uvicorn web server that runs alongside the discord bot. This handles incoming game_result POST requests that will come from the Riot tournament server.

### CODE REVIEW

File: inhouse_bot/database_orm/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-from inhouse_bot.database_orm.session.session_handler import session_scope, bot_declarative_base
+from inhouse_bot.database_orm.session.session import session_scope, bot_declarative_base
```
Comment (adamschachne) [inline inhouse_bot/database_orm/__init__.py:1]: this is just a rename

File: inhouse_bot/common_utils/get_server_config.py [hunk 2]
```diff
@@ -1,6 +1,6 @@
 from inhouse_bot.common_utils.constants import CONFIG_OPTIONS
 from inhouse_bot.database_orm import ServerConfig
-from inhouse_bot.database_orm.session.session_handler import session_scope
+from inhouse_bot.database_orm.session.session import session_scope
```
Comment (adamschachne) [inline inhouse_bot/common_utils/get_server_config.py:3]: rename only

File: inhouse_bot/cogs/queue_cog.py [hunk 3]
```diff
@@ -95,7 +95,7 @@ async def run_matchmaking_logic(
 
             if ready is True:
                 # We drop all 10 players from the queue
-                game_queue.validate_ready_check(ready_check_message.id)
+                game_queue.remove_players_from_queue(ready_check_message.id)
```
Comment (adamschachne) [inline inhouse_bot/cogs/queue_cog.py:98]: rename

File: inhouse_bot/common_utils/lol_api/tasks.py [hunk 4]
```diff
@@ -23,4 +23,3 @@ async def average_win_rate_10_matches(summoner_name: str):
             if participant.puuid == summoner.puuid:
                 wins.append(int(participant.win))
     return statistics.mean(wins or [0])
```
Comment (adamschachne) [inline inhouse_bot/common_utils/lol_api/tasks.py:25]: format

File: inhouse_bot/game_queue/__init__.py [hunk 5]
```diff
@@ -6,7 +6,7 @@
     remove_player,
     remove_players,
     start_ready_check,
-    validate_ready_check,
+    remove_players_from_queue,
```
Comment (adamschachne) [inline inhouse_bot/game_queue/__init__.py:9]: rename

File: inhouse_bot/game_queue/queue_handler.py [hunk 6]
```diff
@@ -140,7 +140,7 @@ def start_ready_check(player_ids: List[int], channel_id: int, ready_check_messag
         )
 
 
-def validate_ready_check(ready_check_id: int):
+def remove_players_from_queue(ready_check_id: int):
```
Comment (adamschachne) [inline inhouse_bot/game_queue/queue_handler.py:143]: rename

File: inhouse_bot/queue_channel_handler/queue_channel_handler.py [hunk 7]
```diff
@@ -195,7 +195,7 @@ class QueueChannelsOnly(commands.CheckFailure):
 
 # This is a decorator for commands
 def queue_channel_only():
-    async def predicate(ctx):
+    async def predicate(ctx: commands.Context):
```
Comment (adamschachne) [inline inhouse_bot/queue_channel_handler/queue_channel_handler.py:198]: Just adding typing here

File: inhouse_bot/ranking_channel_handler/ranking_channel_handler.py [hunk 8]
```diff
@@ -119,5 +119,4 @@ def get_server_ratings(server_id: int, role: str = None, limit=100):
 
         return ratings
```
Comment (adamschachne) [inline inhouse_bot/ranking_channel_handler/ranking_channel_handler.py:121]: format only

File: tests/game_queue/test_queue_handler.py [hunk 9]
```diff
@@ -57,7 +57,7 @@ def test_queue_full():
 
     # We start and validate the ready check (message id 1)
     game_queue.start_ready_check(list(range(1, 11)), 0, 1)
-    game_queue.validate_ready_check(1)
+    game_queue.remove_players_from_queue(1)
```
Comment (adamschachne) [inline tests/game_queue/test_queue_handler.py:60]: rename only

File: tests/matchmaking_logic/test_matchmaking_logic.py [hunk 10]
```diff
@@ -98,6 +98,6 @@ def test_duo_matchmaking():
             winner = game.player_ids_list[int(random.random() * 10)]
 
         game_queue.start_ready_check([i for i in range(0, 10)], 0, 0)
-        game_queue.validate_ready_check(0)
+        game_queue.remove_players_from_queue(0)
```
Comment (adamschachne) [inline tests/matchmaking_logic/test_matchmaking_logic.py:101]: rename only

File: tests/test_cog.py [hunk 11]
```diff
@@ -111,7 +111,7 @@ async def game(self, ctx: commands.Context):
         msg = await ctx.send("The queue has been reset, filled again, and a game created (with no winner)")
 
         game_queue.start_ready_check([i for i in range(0, 9)] + [ctx.author.id], ctx.channel.id, msg.id)
-        game_queue.validate_ready_check(msg.id)
+        game_queue.remove_players_from_queue(msg.id)
```
Comment (adamschachne) [inline tests/test_cog.py:114]: rename

File: tests/test_cog.py [hunk 12]
```diff
@@ -139,7 +139,7 @@ async def games(self, ctx: commands.Context):
                 winner = game.player_ids_list[int(random.random() * 10)]
 
             game_queue.start_ready_check([i for i in range(0, 9)] + [ctx.author.id], ctx.channel.id, 0)
-            game_queue.validate_ready_check(0)
+            game_queue.remove_players_from_queue(0)
```
Comment (adamschachne) [inline tests/test_cog.py:143]: rename

File: inhouse_bot/database_orm/__init__.py (hunk 1 above)
Comment (adamschachne) [inline inhouse_bot/database_orm/__init__.py:1]: just a rename

**Review by classBabacar — APPROVED** [review]

**Pull request merged**